mars-sim-core/src/main/java/com/mars_sim/core/person/ai/CacheCreator.java

```diff
@@ -0,0 +1,89 @@
+package com.mars_sim.core.person.ai;
+
+import com.mars_sim.core.person.ai.task.util.TaskJob;
+import com.mars_sim.core.time.MarsTime;
+import com.mars_sim.core.tool.RandomUtil;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CacheCreator<T extends TaskJob> {
```

Comment:
Can we expand it to also replace the Mission selection code? We would a new interface, ScoredItem, that has a get score method.
Then we could use the Generics for both cases. Maybe rename the class to ScoredItemChooser.